### PR TITLE
UPSTREAM: 17886: pod log location must validate container if provided

### PR DESCRIPTION
**Pull request by fabianofranz**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1286094

### CODE REVIEW

**Comment by fabianofranz** [comment]
Upstream PR: https://github.com/kubernetes/kubernetes/pull/17886

@ncdc not sure if it's you, but ptal (here and upstream).

File: Godeps/_workspace/src/k8s.io/kubernetes/pkg/registry/pod/strategy.go [hunk 1]
```diff
@@ -235,6 +235,7 @@ func LogLocation(getter ResourceGetter, connInfo client.ConnectionInfoGetter, ct
 	}
 
 	// Try to figure out a container
+	// If a container were provided, it must be valid
```
Comment (liggitt) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/registry/pod/strategy.go]: s/were/was/

Comment (fabianofranz) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/registry/pod/strategy.go]: fixed

**Comment by fabianofranz** [comment]
[test]

**Comment by liggitt** [comment]
```
--- FAIL: TestRegistryResourceLocation (0.00s)
	rest_test.go:73: Status: Complete Expected Location: https://foo-host:12345/containerLogs/default/running-build/foo-container, Got https://foo-host:12345/containerLogs/default/foo-pod/foo-container
	rest_test.go:73: Status: Failed Expected Location: https://foo-host:12345/containerLogs/default/running-build/foo-container, Got https://foo-host:12345/containerLogs/default/foo-pod/foo-container
	rest_test.go:73: Status: Running Expected Location: https://foo-host:12345/containerLogs/default/running-build/foo-container, Got https://foo-host:12345/containerLogs/default/foo-pod/foo-container
```

**Comment by fabianofranz** [comment]
@liggitt Seems just the mock, since we are now using the actual name from the rest api object, the mocks have to honour it. Fixed, let's see how it goes.

**Comment by liggitt** [comment]
Yup, split origin changes into their own commit

**Comment by fabianofranz** [comment]
@liggitt of course, missed that. Amending.

**Comment by smarterclayton** [comment]
I will cut 1.1.0.1 with this once merged.

**Comment by liggitt** [comment]
Lgtm, [merge]

**Comment by sdodson** [comment]
> I will cut 1.1.0.1 with this once merged.

@smarterclayton This still happening?

**Comment by liggitt** [comment]
Should also pick up https://github.com/openshift/origin/pull/6129

**Comment by smarterclayton** [comment]
Once Jordan tells me to

On Dec 1, 2015, at 10:50 AM, Jordan Liggitt <notifications@github.com>
wrote:

Should also pick up #6129 <https://github.com/openshift/origin/pull/6129>

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/6113#issuecomment-161007488>.

**Comment by liggitt** [comment]
@smarterclayton go ahead (either from master or from ba7f5102daecf1668aaedae9c314a25a2d81e593)

**Pull request merged**